### PR TITLE
Decoupling controller

### PR DESCRIPTION
**Pull request by Onesco**
# Description
Decoupling of the initial tightly coupled route endpoints to the data access to loosely coupled. This is done to enable easy integration of other database sources in the future and also easy writing of unit tests for each endpoint.
  
Fixes # (issue)
#13 
## Type of change
Enhancement. 

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Onesco** [comment]
@anomic30 done

File: server/controllers/file.js [hunk 1]
```diff
@@ -0,0 +1,258 @@
+const { createWriteStream, unlink } = require('fs');
```
Comment (anomic30) [inline server/controllers/file.js:1]: Import readFileSync

File: server/controllers/file.js [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+module.exports = class DownloadController{
+    static async secureDownload(req, res){
+        console.log("Secure download called");
+        const cid = req.params.cid;
+        const auth = req.params.auth;
+        const metadata = await magic.users.getMetadataByToken(auth);
+        const magic_id = metadata.issuer;
+    
+        try {
+            const file = await User.findOne({ magic_id: magic_id, files: { $elemMatch: { cid: cid } } }, { encryption_key: 1 }).select({ files: { $elemMatch: { cid: cid } } });
```
Comment (anomic30) [inline server/controllers/file.js:63]: Error while downloading a file: 
image

Maybe modify the user.js file in the data-repository.

File: server/controllers/file.js [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+            }
+        } catch (err) {
+            return res.status(500).json({ error: err.message });
+        }
+    }
+
+    static async download(req, res){
+        const cid = req.params.cid;
+        try {
+            const file = await User.findOne({ files: { $elemMatch: { cid: cid, public: true } } }, { encryption_key: 1 }).select({ files: { $elemMatch: { cid: cid, public: true } } });
```
Comment (anomic30) [inline server/controllers/file.js:121]: Same for here too.
image

**Review by anomic30 — COMMENTED** [review]
Check every API routes.

**Comment by anomic30** [comment]
Hi @Onesco, check and fix all the API routes from the frontend and raise a PR :)

**Comment by Onesco** [comment]
@anomic30 alright

**Pull request merged**